src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/FunctionItemCallIterator.java

```diff
@@ -225,8 +243,15 @@ public void setNextResult() {
 
     @Override
     public JavaRDD<Item> getRDDAux(DynamicContext dynamicContext) {
-        processArguments();
-        return _functionBodyIterator.getRDD(_currentDynamicContext);
+        if (_isPartialApplication) {
+            throw new OurBadException(
```

Comment:
Yes, exactly -- in the static analysis, you can simply make sure that all partial function evaluations are locally evaluated.

This makes sense, since to the best of my understand, only one item is then returned anyway: using RDDs would be an unnecessary overhead.